app/views/contexts/summary/group.py

```diff
@@ -56,19 +62,20 @@ def __init__(
     def _build_blocks_and_links(
         self,
         *,
-        group_schema,
-        routing_path,
-        answer_store,
-        list_store,
-        metadata,
-        response_metadata,
-        schema,
-        location,
-        return_to,
-        progress_store,
-        language,
-        return_to_block_id,
-    ):
+        group_schema: Mapping[str, Any],
+        routing_path: RoutingPath,
+        answer_store: AnswerStore,
+        list_store: ListStore,
+        metadata: Optional[MetadataProxy],
+        response_metadata: Mapping,
+        schema: QuestionnaireSchema,
+        location: Location,
+        return_to: Optional[str],
+        progress_store: ProgressStore,
+        language: str,
+        return_to_block_id: Optional[str],
+    ) -> list[Mapping[str, Block]]:
```

Comment:
This should just be a list of dicts.

Reply:
Actually leave it as is. Although dict is more concrete, Mapping is correct if we don't want to allow mutation

Reply:
👍